test/smoke/gateway_single_spoke_test.go

```diff
@@ -0,0 +1,267 @@
+package smoke
```

Comment:
```suggestion
//go:build e2e

package smoke
```

I think it won't make a difference practically but just to follow the convention introduced in #244 I'd suggest to add the tag to separate these from the integration tests